indexing-hadoop/src/main/java/io/druid/indexer/updater/HadoopConverterJob.java

```diff
@@ -0,0 +1,995 @@
+        segmentLocURI = URI.create(String.format("s3n://%s/%s", loadSpec.get("bucket"), loadSpec.get("key")));
+      } else if ("hdfs".equals(type)) {
+        segmentLocURI = URI.create(loadSpec.get("path").toString());
+      } else if ("local".equals(type)) {
+        try {
+          segmentLocURI = new URI("file", null, loadSpec.get("path").toString(), null, null);
+        }
+        catch (URISyntaxException e) {
+          throw new ISE(e, "Unable to form simple file uri");
+        }
+      } else {
+        try {
+          throw new IAE(
+              "Cannot figure out loadSpec %s",
+              HadoopDruidConverterConfig.jsonMapper.writeValueAsString(loadSpec)
+          );
+        }
+        catch (JsonProcessingException e) {
+          throw new ISE("Cannot write Map with json mapper");
+        }
+      }
+      return segmentLocURI;
+    }
+
+    @Override
+    protected void cleanup(
+        Context context
+    ) throws IOException, InterruptedException
+    {
+      final String tmpDirLoc = context.getConfiguration().get(TMP_FILE_LOC_KEY);
+      final File tmpDir = Paths.get(tmpDirLoc).toFile();
+      FileUtils.deleteDirectory(tmpDir);
+      context.progress();
+      context.setStatus("Clean");
+    }
+  }
+
+  public static class DataSegmentSplitSerializer implements Serialization<DataSegmentSplit>
+  {
+    @Override
+    public boolean accept(Class<?> c)
+    {
+      return DataSegmentSplit.class.isAssignableFrom(c);
+    }
+
+    @Override
+    public Serializer<DataSegmentSplit> getSerializer(Class<DataSegmentSplit> c)
+    {
+      return new Serializer<DataSegmentSplit>()
+      {
+        OutputStream out;
+
+        @Override
+        public void open(OutputStream out) throws IOException
+        {
+          this.out = out;
+        }
+
+        @Override
+        public void serialize(DataSegmentSplit dataSegmentSplit) throws IOException
+        {
+          out.write(HadoopDruidConverterConfig.jsonMapper.writeValueAsBytes(dataSegmentSplit.dataSegment));
+        }
+
+        @Override
+        public void close() throws IOException
+        {
+          out.close();
+        }
+      };
+    }
+
+    @Override
+    public Deserializer<DataSegmentSplit> getDeserializer(Class<DataSegmentSplit> c)
+    {
+      return new Deserializer<DataSegmentSplit>()
+      {
+        InputStream stream;
+
+        @Override
+        public void open(InputStream in) throws IOException
+        {
+          this.stream = in;
+        }
+
+        @Override
+        public DataSegmentSplit deserialize(DataSegmentSplit dataSegmentSplit)
+            throws IOException
+        {
+          return new DataSegmentSplit(HadoopDruidConverterConfig.jsonMapper.readValue(stream, DataSegment.class));
+        }
+
+        @Override
+        public void close() throws IOException
+        {
+          stream.close();
+        }
+      };
+    }
+  }
+
+  public static class DataSegmentSplit extends InputSplit
```

Comment:
it is more conventional to implement "org.apache.hadoop.io.Writable" as well so that you don't have to create and setup serde separately. I think that will reduce some code(DataSegmentSplitSerializer) and setupSerializers(..) method.